### PR TITLE
[AIRFLOW-5801] Get GCP credentials from file instead of JSON blob

### PR DESCRIPTION
**Pull request by ephraimbuddy**
---
This PR fixes 'CloudSqlProxyRunner always assumes GCP credentials are passed via file instead of JSON blob'
https://issues.apache.org/jira/browse/AIRFLOW-5801

Issue link: [AIRFLOW-5801](https://issues.apache.org/jira/browse/AIRFLOW-5801)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/cloud_sql.py [hunk 1]
```diff
@@ -520,12 +520,12 @@ def _get_credential_parameters(self, session: Session) -> List[str]:
...
             filter(Connection.conn_id == self.gcp_conn_id).first()
         session.expunge_all()
-        if GCP_CREDENTIALS_KEY_PATH in connection.extra_dejson:
+        if connection.extra_dejson.get(GCP_CREDENTIALS_KEY_PATH):
             credential_params = [
                 '-credential_file',
                 connection.extra_dejson[GCP_CREDENTIALS_KEY_PATH]
             ]
-        elif GCP_CREDENTIALS_KEYFILE_DICT in connection.extra_dejson:
+        elif connection.extra_dejson.get(GCP_CREDENTIALS_KEYFILE_DICT):
```
Comment (turbaszek) [inline airflow/providers/google/cloud/hooks/cloud_sql.py:528]: Is this change necessary? Personally I think the current version is more accurate

File: tests/providers/google/cloud/hooks/test_cloud_sql.py [hunk 2]
```diff
@@ -1143,7 +1143,7 @@ def test_cloudsql_database_hook_get_database_hook(self, get_connection):
 
 class TestCloudSqlDatabaseHook(unittest.TestCase):
 
-    @mock.patch('airflow.contrib.hooks.gcp_sql_hook.CloudSQLDatabaseHook.get_connection')
+    @mock.patch('airflow.providers.google.cloud.hooks.cloud_sql.CloudSQLDatabaseHook.get_connection')
```
Comment (turbaszek) [inline tests/providers/google/cloud/hooks/test_cloud_sql.py:1146]: How this change is related to PR title?

File: airflow/providers/google/cloud/hooks/cloud_sql.py (hunk 1 above)
Comment (ephraimbuddy) [reply airflow/providers/google/cloud/hooks/cloud_sql.py:528]: I was trying to resolve this issue https://issues.apache.org/jira/browse/AIRFLOW-5801.

File: tests/providers/google/cloud/hooks/test_cloud_sql.py (hunk 2 above)
Comment (ephraimbuddy) [reply tests/providers/google/cloud/hooks/test_cloud_sql.py:1146]: Thanks for pointing this out, it will help me to improve.
After running the test, I saw deprecation warning and decided to fix it. 
I know now that I should have created an issue for it or make a better PR title.
I really appreciate 🙂

File: airflow/providers/google/cloud/hooks/cloud_sql.py (hunk 1 above)
Comment (mik-laj) [reply airflow/providers/google/cloud/hooks/cloud_sql.py:528]: @tomek these elements exist but are empty when you create JSON through the form in the web UI.

File: tests/providers/google/cloud/hooks/test_cloud_sql.py (hunk 2 above)
Comment (mik-laj) [reply tests/providers/google/cloud/hooks/test_cloud_sql.py:1146]: +1  It looks good. We should delete references to airflow.contrib.

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Looks good. Just in time to release it in backport operators!

File: airflow/providers/google/cloud/hooks/cloud_sql.py (hunk 1 above)
Comment (turbaszek) [reply airflow/providers/google/cloud/hooks/cloud_sql.py:528]: Oh, got it!

**Review by turbaszek — APPROVED** [review]

**Review by mik-laj — APPROVED** [review]
